lib/galaxy/exceptions/__init__.py

```diff
@@ -221,6 +221,11 @@ class InvalidFileFormatError(MessageException):
     err_code = error_codes_by_name['INVALID_FILE_FORMAT']
 
 
+class ReferenceDataError(MessageException):
+    status_code = 500
+    err_code = error_codes_by_name['REFERENCE_DATA_ERROR']
```

Comment:
Sounds more like an invalid request ? 500 doesn't seem right here

Reply:
I don't mind changing it, but, I thought it made more sense as a server-side error. So, just in case, to double check, here's my reasoning:
Here's an example of an invalid request: https://github.com/galaxyproject/galaxy/blob/16a420c9660819a3f7c15e7775bd470f42b80d1e/lib/galaxy/managers/genomes.py#L54 - i.e., the supplied index type is invalid.
But in the all cases where I use `ReferenceDataError` it is on the server end: the key is valid, but Galaxy can't find the reference data for that key, or can't find any reference data for that index type, etc. - which may happen for various reasons:
https://github.com/galaxyproject/galaxy/blob/16a420c9660819a3f7c15e7775bd470f42b80d1e/lib/galaxy/managers/genomes.py#L63 :point_up: also see line 70 and 71; 
and here:
https://github.com/galaxyproject/galaxy/blob/16a420c9660819a3f7c15e7775bd470f42b80d1e/lib/galaxy/visualization/genomes.py#L95 :point_up: and line 375.

If this can be treated as invalid requests, I'll change it to `RequestParameterInvalidException`.

Reply:
Alright, that makes sense. Thanks @ic4f!